File: Retos/Reto #0 - EL FAMOSO FIZZ BUZZ [Fácil]/typescript/MarckWeb.ts
/*
 * Escribe un programa que muestre por consola (con un print) los
 * números de 1 a 100 (ambos incluidos y con un salto de línea entre
 * cada impresión), sustituyendo los siguientes:
 * - Múltiplos de 3 por la palabra "fizz".
 * - Múltiplos de 5 por la palabra "buzz".
 * - Múltiplos de 3 y de 5 a la vez por la palabra "fizzbuzz".
 */

const printfizzBuzz = () => {
   for (let i: number = 0; i < 101; i++) {
      let output: string = ''
      if (i % 3 == 0) output = 'fizz'
      if (i % 5 == 0) output = 'buzz'
      if (i % 3 == 0 && i % 5 == 0) output = 'fizzbuzz'

      console.log(`${i}= ${output}`)
   }
}

printfizzBuzz()

File: Retos/Reto #0 - EL FAMOSO FIZZ BUZZ [Fácil]/python/MarckWeb.py
""" 
/*
 * Escribe un programa que muestre por consola (con un print) los
 * números de 1 a 100 (ambos incluidos y con un salto de línea entre
 * cada impresión), sustituyendo los siguientes:
 * - Múltiplos de 3 por la palabra "fizz".
 * - Múltiplos de 5 por la palabra "buzz".
 * - Múltiplos de 3 y de 5 a la vez por la palabra "fizzbuzz".
*/ 
"""

# Ejercicio con condicion ternario


def fizzBuzz():
    for i in range(101):
        result = 'Fizz' * (i % 3 == 0) + 'Buzz' * (i % 5 == 0)
        print(result if result else i)


fizzBuzz()

# Ejercicio con condicion if, elif


def fizzBuzz():
    for i in range(101):
        output = ''
        divisibleForTHree = i % 3 == 0
        divisibleForFive = i % 5 == 0
        if divisibleForTHree and divisibleForFive:
            output = 'FizzBuzz'
        elif divisibleForTHree:
            output = 'Fizz'
        elif divisibleForFive:
            output = 'Buzz'
        else:
            output = i
        print(output)


fizzBuzz()
